test/suite/test_checkpoint_snapshot02.py

```diff
@@ -0,0 +1,136 @@
+from wtscenario import make_scenarios
+from wiredtiger import stat
+
+# test_checkpoint_snapshot02.py
+#   This test is to run checkpoint and eviction in parallel with timing
+#   stress for checkpoint and let eviction write more data than checkpoint.
+#
+
+class test_checkpoint_snapshot02(wttest.WiredTigerTestCase):
+
+    # Create a table.
+    uri = "table:test_checkpoint_snapshot02"
+    nrows = 200
+
+    def conn_config(self):
+        config = 'cache_size=5MB,statistics=(all),statistics_log=(json,on_close,wait=1),log=(enabled=true),timing_stress_for_test=[checkpoint_slow]'
+        return config
+
+    def large_updates(self, uri, value, ds, nrows):
+        # Update a large number of records.
+        session = self.session
+        cursor = session.open_cursor(uri)
+        for i in range(0, nrows):
+            session.begin_transaction()
+            cursor[ds.key(i)] = value
+            session.commit_transaction()
+        cursor.close()
+
+    def check(self, check_value, uri, nrows):
+        session = self.session
+        session.begin_transaction()
+        cursor = session.open_cursor(uri)
+        count = 0
+        for k, v in cursor:
+            self.assertEqual(v, check_value)
+            count += 1
+        session.commit_transaction()
+        self.assertEqual(count, nrows)
+
+    def test_checkpoint_snapshot(self):
+
+        #ds = SimpleDataSet(self, self.uri, self.nrows, key_format="S", value_format='u')
+        ds = SimpleDataSet(self, self.uri, 0, key_format="S", value_format="S")
+        ds.populate()
+        valuea = "aaaaa" * 100
+        valueb = "bbbbb" * 100
+        valuec = "ccccc" * 100
+        valued = "ddddd" * 100
+
+        cursor = self.session.open_cursor(self.uri)
+        self.session.begin_transaction()
+
+        # Create a checkpoint thread
+        done = threading.Event()
+        ckpt = checkpoint_thread(self.conn, done)
+        try:
+            self.pr("start checkpoint thread")
+            ckpt.start()
+
+            # Insert some data from the transaction which is running before
+            # checkpoint started
+            for i in range(0, self.nrows):
+                cursor.set_key(ds.key(i))
+                cursor.set_value(valuea)
+                self.assertEqual(cursor.insert(), 0)
+            self.session.commit_transaction()
+
+            self.large_updates(self.uri, valueb, ds, self.nrows)
+            self.large_updates(self.uri, valuec, ds, self.nrows)
+            self.large_updates(self.uri, valued, ds, self.nrows)
+
+        finally:
+            done.set()
+            ckpt.join()
+
+        #Simulate a crash by copying to a new directory(RESTART).
+        copy_wiredtiger_home(".", "RESTART")
+
+        # Open the new directory.
+        self.conn = self.setUpConnectionOpen("RESTART")
+        self.session = self.setUpSessionOpen(self.conn)
+
+        # Check the table contains the last checkpointed value.
+        self.check(valued, self.uri, self.nrows)
```

Comment:
The WT checkpoint threading function continuously performs checkpoint while the updates are getting inserted. I am not sure whether we can get a reliable scenario of eviction writing more than a checkpoint. As the test is passing when you are comparing the value with the latest insert that is done in parallel to the checkpoint. I think the table doesn't have any data for the first checkpoint, due to which it happened. Can you perform large updates with `valuea` first before triggering the checkpoint?

Reply:
Updated the test to insert large updates before the checkpoint and looks like inconsistent checkpoint behavior is triggered.